frontend/src/lib/logic/inAppPrompt/inAppPromptLogic.tsx

```diff
@@ -44,27 +44,21 @@ export type Prompt = {
     type: PromptType
     text: string
     placement: Placement
-    reference: string
+    reference: string | null
     title?: string
     buttons?: PromptButton[]
     icon?: string
 }
 
 export type Tooltip = Prompt & { type: 'tooltip' }
 
-export type PromptRule = {
-    path: {
-        must_match: string[]
-        exclude?: string[]
-    }
-    must_be_completed?: string[]
-    requires_opt_in?: boolean
-}
-
 export type PromptSequence = {
     key: string
     prompts: Prompt[]
-    rule: PromptRule
+    path_match: string[]
```

Comment:
For reviewers, most of this file is just updated to match the new schema, and tests have been updated accordingly, there is no new logic